pkg/aws/cloud.go

```diff
@@ -42,24 +43,25 @@ type Cloud interface {
 
 // NewCloud constructs new Cloud implementation.
 func NewCloud(cfg CloudConfig, metricsRegisterer prometheus.Registerer) (Cloud, error) {
-	metadataSess := session.Must(session.NewSession(aws.NewConfig()))
-	metadata := services.NewEC2Metadata(metadataSess)
-	if len(cfg.Region) == 0 {
-		region, err := metadata.Region()
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to introspect region from EC2Metadata, specify --aws-region instead if EC2Metadata is unavailable")
-		}
-		cfg.Region = region
-	}
 
 	if len(cfg.VpcID) == 0 {
+		metadataSess := session.Must(session.NewSession(aws.NewConfig()))
+		metadata := services.NewEC2Metadata(metadataSess)
 		vpcId, err := metadata.VpcID()
 		if err != nil {
 			return nil, errors.Wrap(err, "failed to introspect vpcID from EC2Metadata, specify --aws-vpc-id instead if EC2Metadata is unavailable")
 		}
 		cfg.VpcID = vpcId
 	}
 
+	if len(cfg.Region) == 0 {
+		region := os.Getenv("AWS_DEFAULT_REGION")
```

Comment:
These env variables are set only if the IRSA is configured. We still need metadata introspection for the region configuration as a fallback.